platforms/core-execution/execution/src/test/groovy/org/gradle/internal/execution/history/changes/MessageCollectingChangeVisitorsTest.groovy

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.execution.history.changes
+
+import spock.lang.Specification
+
+class MessageCollectingChangeVisitorsTest extends Specification {
```

Comment:
I noticed that if I named this class `MessageCollectingChangeVisitorTest` in `MessageCollectingChangeVisitorTest.groovy` (what I wanted to call it) then it didn't report any failures and probably didn't run, so I added an `s` into the name of the test to get it to run. Does this seem like something we should be concerned about?

Reply:
The name of the class should match the file name, but even if that's not the case, the tests should run. I tried renaming a similar test class and the tests in it kept executing fine, as they should.

Reply:
I think I found a repro for the problem I was encountering about the test not running: https://github.com/gradle/gradle/issues/28916

In the meanwhile it appears that if I delete my Gradle cache that the test starts running again, so I renamed the test and its test file back to what I originally wanted to call it